File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/dtos/req/CoffeeBeanAvgExtractionRating.java
package at.ac.tuwien.sepm.groupphase.backend.dtos.req;

public class CoffeeBeanAvgExtractionRating {
    private Long id;
    private Double body;
    private Double acidity;
    private Double aromatics;
    private Double sweetness;
    private Double aftertaste;

    public CoffeeBeanAvgExtractionRating() {
    }

    public CoffeeBeanAvgExtractionRating(Long id, Double body, Double acidity, Double aromatics, Double sweetness, Double aftertaste) {
        this.id = id;
        this.body = body;
        this.acidity = acidity;
        this.aromatics = aromatics;
        this.sweetness = sweetness;
        this.aftertaste = aftertaste;
    }

    public Long getId() {
        return id;
    }

    public Double getBody() {
        return body;
    }

    public Double getAcidity() {
        return acidity;
    }

    public Double getAromatics() {
        return aromatics;
    }

    public Double getSweetness() {
        return sweetness;
    }

    public Double getAftertaste() {
        return aftertaste;
    }
}

File: frontend/src/dtos/req/coffee-bean-avg-extraction-rating.ts
export class CoffeeBeanAvgExtractionRating {
  id: number;
  body: number;
  acidity: number;
  aromatics: number;
  sweetness: number;
  aftertaste: number;

  constructor(
    id: number,
    body: number,
    acidity: number,
    aromatics: number,
    sweetness: number,
    aftertaste: number
  ) {
    this.id = id;
    this.body = body;
    this.acidity = acidity;
    this.aromatics = aromatics;
    this.sweetness = sweetness;
    this.aftertaste = aftertaste;
  }

  getChartData(): number[] {
    return [
      this.body,
      this.acidity,
      this.aromatics,
      this.sweetness,
      this.aftertaste,
    ];
  }
}
